community/cypher/cypher-compiler-3.0/src/test/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/plans/IndexScanLeafPlannerTest.scala

```diff
@@ -38,120 +38,224 @@ class IndexScanLeafPlannerTest extends CypherFunSuite with LogicalPlanningTestSu
-    }.withLogicalPlanningContext { (cfg, ctx) =>
-      // when
-      val resultPlans = indexScanLeafPlanner(cfg.qg)(ctx)
+        uniqueIndexOn("Awesome", "prop")
+      }.withLogicalPlanningContext { (cfg, ctx) =>
+        // when
+        val resultPlans = indexScanLeafPlanner(cfg.qg)(ctx)
 
-      // then
-      resultPlans should beLike {
-        case Seq(NodeIndexScan(`idName`, _, _, _)) => ()
+        // then
+        resultPlans should beLike {
+          case Seq(NodeIndexScan(`idName`, _, _, _)) => ()
+        }
       }
     }
-  }
 
-  test("plans index scans such that it solves hints") {
-    val hint: UsingIndexHint = UsingIndexHint(varFor("n"), LabelName("Awesome")_, PropertyKeyName("prop")(pos))_
+    test("plans index scans such that it solves hints") {
+      val hint: UsingIndexHint = UsingIndexHint(varFor("n"), LabelName("Awesome") _, PropertyKeyName("prop")(pos)) _
 
-    new given {
-      qg = queryGraph(existsPredicate, hasLabels).addHints(Some(hint))
+      new given {
+        qg = queryGraph(existsPredicate, hasLabels).addHints(Some(hint))
 
-      indexOn("Awesome", "prop")
-    }.withLogicalPlanningContext { (cfg, ctx) =>
-      // when
-      val resultPlans = indexScanLeafPlanner(cfg.qg)(ctx)
+        indexOn("Awesome", "prop")
+      }.withLogicalPlanningContext { (cfg, ctx) =>
+        // when
+        val resultPlans = indexScanLeafPlanner(cfg.qg)(ctx)
 
-      // then
-      resultPlans should beLike {
-        case Seq(NodeIndexScan(`idName`, _, _, _)) => ()
+        // then
+        resultPlans should beLike {
+          case Seq(NodeIndexScan(`idName`, _, _, _)) => ()
+        }
+
+        resultPlans.map(_.solved.queryGraph) should beLike {
+          case (Seq(plannedQG: QueryGraph)) if plannedQG.hints == Set(hint) => ()
+        }
       }
+    }
+
+    test("plans unique index scans such that it solves hints") {
+      val hint: UsingIndexHint = UsingIndexHint(varFor("n"), LabelName("Awesome") _, PropertyKeyName("prop")(pos)) _
+
+      new given {
+        qg = queryGraph(existsPredicate, hasLabels).addHints(Some(hint))
+
+        uniqueIndexOn("Awesome", "prop")
+      }.withLogicalPlanningContext { (cfg, ctx) =>
+        // when
+        val resultPlans = indexScanLeafPlanner(cfg.qg)(ctx)
 
-      resultPlans.map(_.solved.queryGraph) should beLike {
-        case (Seq(plannedQG: QueryGraph)) if plannedQG.hints == Set(hint) => ()
+        // then
+        resultPlans should beLike {
+          case Seq(NodeIndexScan(`idName`, _, _, _)) => ()
+        }
+
+        resultPlans.map(_.solved.queryGraph) should beLike {
+          case (Seq(plannedQG: QueryGraph)) if plannedQG.hints == Set(hint) => ()
+        }
       }
     }
-  }
 
-  test("plans unique index scans such that it solves hints") {
-    val hint: UsingIndexHint = UsingIndexHint(varFor("n"), LabelName("Awesome")_, PropertyKeyName("prop")(pos))_
+    test("plans index scans for: n.prop STARTS WITH <pattern>") {
```

Comment:
If the aim is to group by operator, there should be a comment `TESTS OF STARTS WITH` here.